activerecord/test/cases/adapters/postgresql/transaction_nested_test.rb

```diff
@@ -98,8 +102,10 @@ class Bit < ActiveRecord::Base
         with_warning_suppression do
           start_right.wait
           Sample.transaction(isolation: :serializable, requires_new: false) do
+            make_parent_transaction_dirty
```

Comment:
This tests started to fail after this change which either means that the expectation is incorrect or the actual Rails behaviour is incorrect. I'm going to dig into this but meanwhile I've created a separate PR for mysql changes only - https://github.com/rails/rails/pull/44690